src/iotHubResourceExplorer.ts

```diff
@@ -262,9 +262,16 @@ export class IoTHubResourceExplorer extends BaseExplorer {
             if (azureSubscription) {
                 return new SubscriptionItem(azureSubscription.subscription, azureSubscription.session);
             }
+            return undefined;
+        }
+
+        let subscriptionItems = await this.loadSubscriptionItems(this.accountApi);
+        if (subscriptionItems.length === 0) {
+            await vscode.window.showInformationMessage("No subscription found, click an Azure account at the bottom left corner and choose Select All.");
+            return undefined;
         } else {
             const subscriptionItem = await vscode.window.showQuickPick(
-                this.loadSubscriptionItems(this.accountApi),

```

Comment:
`loadSubscriptionItems` would take long time at the first time. So we pass this Promise to the showQuickPick api, so that the loading status would be shown. Therefore, we'd better keep the Promise here. You could consider updating the `loadSubscriptionItems` instead.